File: 0349_intersection_of_two_arrays/Solution.swift
// LeetCode 0349 - Intersection of Two Arrays
// https://leetcode.com/problems/intersection-of-two-arrays/

class Solution {
    func intersection(_ nums1: [Int], _ nums2: [Int]) -> [Int] {
        let set2 = Set(nums2)
        var seen = Set<Int>()
        var result: [Int] = []

        for num in nums1 {
            if set2.contains(num) && !seen.contains(num) {
                seen.insert(num)
                result.append(num)
            }
        }

        return result
    }
}

File: 0349_intersection_of_two_arrays/solution.php
<?php
// LeetCode 0349 - Intersection of Two Arrays
// https://leetcode.com/problems/intersection-of-two-arrays/

class Solution {
    /**
     * @param Integer[] $nums1
     * @param Integer[] $nums2
     * @return Integer[]
     */
    function intersection($nums1, $nums2) {
        $set2 = array_fill_keys($nums2, true);
        $seen = [];
        $result = [];

        foreach ($nums1 as $num) {
            if (isset($set2[$num]) && !isset($seen[$num])) {
                $seen[$num] = true;
                $result[] = $num;
            }
        }

        return $result;
    }
}
